test/e2e/proxy_test.go

```diff
@@ -1084,3 +1078,152 @@ func testProxyServiceLifeCycle(ipFamily *corev1.IPFamily, ingressIPs []string, d
 		}
 	}
 }
+
+// TestProxyLoadBalancerModeDSR creates a LoadBalancer Service and verifies both external and internal clients accessing
+// its LoadBalancer IPs work as expected.
+// Client IP should always be preserved regardless of whether the traffic is externally or internally originated.
+// Session affinity should take effect when it's enabled.
+func TestProxyLoadBalancerModeDSR(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 3)
+	skipIfAntreaIPAMTest(t)
+	skipIfProxyDisabled(t)
+
+	data, err := setupTest(t)
+	require.NoError(t, err, "Error when setting up test")
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+	skipIfEncapModeIsNot(t, data, config.TrafficEncapModeEncap)
```

Comment:
I think you made some changes to the conditions here? I may be missing something, but should there be a "skip" if the DSR Feature Gate is disabled?

Reply:
Fixed. Thanks for catching it. There was no issue because the test was also skipped when proxyAll is disabled.